src/inc/conpty-universal.h

```diff
@@ -30,18 +30,161 @@
 
 const unsigned int PTY_SIGNAL_RESIZE_WINDOW = 8u;
 
-HRESULT CreateConPty(const std::wstring& cmdline,       // _In_
-                     const unsigned short w,            // _In_
-                     const unsigned short h,            // _In_
-                     HANDLE* const hInput,              // _Out_
-                     HANDLE* const hOutput,             // _Out_
-                     HANDLE* const hSignal,             // _Out_
-                     PROCESS_INFORMATION* const piPty); // _Out_
+// A case-insensitive wide-character map is used to store environment variables
+// due to documented requirements:
+//
+//      "All strings in the environment block must be sorted alphabetically by name.
+//      The sort is case-insensitive, Unicode order, without regard to locale.
+//      Because the equal sign is a separator, it must not be used in the name of
+//      an environment variable."
+//      https://docs.microsoft.com/en-us/windows/desktop/ProcThread/changing-environment-variables
+//
+struct WStringCaseInsensitiveCompare
+{
+    bool operator()(const std::wstring& lhs, const std::wstring& rhs) const
+    {
+        return (::_wcsicmp(lhs.c_str(), rhs.c_str()) < 0);
+    }
+};
+
+using EnvironmentVariableMapW = std::map<std::wstring, std::wstring, WStringCaseInsensitiveCompare>;
+
+HRESULT CreateConPty(const std::wstring& cmdline,                       // _In_
+                     const unsigned short w,                            // _In_

```

Comment:
The comments describing `_In_` and `_Out_` are unnecessary, we get that information by virtue of the consts on the params

Reply:
ok, was just copying what was done above

Reply:
@zadjii-msft what is the preferred style here?

Reply:
Eh, pull them or leave them, I don't think it really matters anymore. I think they're vestigial from an earlier version of the API, before we were using consts on those params, and because we had distributed an earlier version of this file to some partners.